js/queen.js

```diff
@@ -0,0 +1,117 @@
+import { Piece } from "./piece.js"
+
+/*
+* Represents a queen chess piece that is used with the Board class
+*/
+export class Queen extends Piece{
+    /**
+     * Instantiates a queen piece with the given position and piece ID.
+     * The piece belongs to the player with the given player ID.
+     * 
+     * @param {int} id Piece's ID.
+     * @param {int} rowPos Piece's row position.
+     * @param {int} colPos Piece's column position.
+     * @param {int} playerOwner
+     * 
+     */
+    constructor(id, rowPos, colPos, playerOwner) {
+        super();
+        this.id = id;
+        this.rowPos = rowPos;
+        this.colPos = colPos;
+        this.playerOwner = playerOwner;
+        this.unicodeChar = "&#9813;";
+    }
+
+    /**
+     * Gets list of possible moves that the queen piece can make.
+     * It looks in every direction the queen can move and continues to
+     * add moves to the possible moves array unless another piece or
+     * the end of the board in encountered. If another piece is encountered,
+     * the queen can only move there if it belongs to the other player.
+     * 
+     * @param {Array.<Array.<int>>} board A 2D array representation of the board.
+     * @returns {Array.<Array.<int>>} Returns a 2D array of the possible moves.
+     */
+    possibleMoves(board) {
+        let possibleMoves = [];
+
+        let maxRows = board.length;
+        let maxCols = board[0].length;
+        let minRows, minCols = 0;
+        /**
+         * Get row increasing moves
+         */
+        for (let i = rowPos + 1; i < maxRows; i++) {
+            if (board[i][colPos] !== null && board[i][colPos].getPlayerOwner() == playerOwner) {
```

Comment:
You'll need to change each call of `getPlayerOwner()` to `getPlayerId()` since the method has been changed in the Piece class.

Reply:
Ever call to getPlayerOwner() is now a call to getPlayerId().